File: src/main/java/Back/bailAutoComplet/BailAutoComplet/Controller/GenerationController.java
package Back.bailAutoComplet.BailAutoComplet.Controller;

import Back.bailAutoComplet.BailAutoComplet.Service.GenerationService;
import Back.bailAutoComplet.BailAutoComplet.model.Generation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/generation")
@CrossOrigin(origins = "http://localhost:4200") // Allowing typical angular port, can be adjusted
public class GenerationController {

    private final GenerationService generationService;

    @Autowired
    public GenerationController(GenerationService generationService) {
        this.generationService = generationService;
    }

    @PostMapping
    public ResponseEntity<Generation> createGeneration(@RequestBody Generation generation) {
        Generation saved = generationService.saveGeneration(generation);
        return ResponseEntity.ok(saved);
    }

    @GetMapping
    public ResponseEntity<List<Generation>> getAllGenerations() {
        return ResponseEntity.ok(generationService.getAllGenerations());
    }
}

File: src/generation/generation.controller.ts
import { Body, Controller, Get, Post } from '@nestjs/common';
import { CreateGenerationDto } from './dto/create-generation.dto';
import { Generation } from './generation.entity';
import { GenerationService } from './generation.service';

@Controller('generation')
export class GenerationController {
  constructor(private readonly generationService: GenerationService) {}

  @Post()
  createGeneration(@Body() body: CreateGenerationDto): Promise<Generation> {
    return this.generationService.saveGeneration(body);
  }

  @Get()
  getAllGenerations(): Promise<Generation[]> {
    return this.generationService.getAllGenerations();
  }
}
